### ISSUE TITLE
`QDrawer`'s animation prevents correct calculation on the `autogrow` property of `QInput`

### ISSUE THREAD

**Issue by maarteNNNN** [quasarframework/quasar#13317]
### What happened?

Using a `QInput` inside `QDrawer` with the `autogrow` property fails to calculate the right height in [`adjustHeight`](https://github.com/quasarframework/quasar/blob/6fe4a89c6fee5ad89aada5456a06358ae25081c4/ui/src/components/input/QInput.js#L285-L301) when using `v-show`. If using a `v-if` the problem is resolved. However in my specific use case this is a component that has functions which need to be listened to. So the component needs to be mounted.
![image](https://user-images.githubusercontent.com/14275291/166233230-6ef789b8-60e6-4bee-bc40-8ed7bab65661.png)

### What did you expect to happen?

Autogrow would grow correctly. I presume that due to the drawers animation the element isn't yet populated with the correct height properties, thus it calculating the `height: 0px`. I think an event should be fired once the Drawer's animation is completed to recalculate the height. I simulated this via forcing a `v-if` on the input which are determined by the `@hide` and `@show` properties of the QDrawer [see this example here](https://codepen.io/maarteNNNN/pen/oNENKem). Then it's correctly calculating the height.

### Reproduction URL

https://codepen.io/maarteNNNN/pen/OJQJKPL

### How to reproduce?

Just open the drawer on an initial load. Once a firing a digit the field grows correctly and the problem is resolved. Even upon closing. So be sure to always reload to see the bug.

### Flavour

Quasar CLI with Webpack (@quasar/cli | @quasar/app-webpack)

### Areas

Components (quasar)

### Platforms/Browsers

Chrome

### Quasar info output

```Shell
Operating System - Linux(5.16.19-76051619-generic) - linux/x64
NodeJs - 14.17.4

Global packages
  NPM - 8.5.3
  yarn - 1.22.17
  @quasar/cli - 1.3.2
  @quasar/icongenie - Not installed
  cordova - Not installed

Important local packages
  quasar - 2.6.6 -- Build high-performance VueJS user interfaces (SPA, PWA, SSR, Mobile and Desktop) in record time
  @quasar/app - 3.3.3 -- Quasar Framework local CLI
  @quasar/extras - 1.13.6 -- Quasar Framework fonts, icons and animations
  eslint-plugin-quasar - Not installed
  vue - 3.2.33 -- The progressive JavaScript framework for building modern web UI.
  vue-router - 4.0.14
  vuex - 4.0.2 -- state management for Vue.js
  electron - Not installed
  electron-packager - Not installed
  electron-builder - Not installed
  @babel/core - 7.17.10 -- Babel compiler core.
  webpack - 5.72.0 -- Packs CommonJs/AMD modules for the browser. Allows to split your codebase into multiple bundles, which can be loaded on demand. Support loaders to preprocess files, i.e. json, jsx, es7, css, less, ... and your custom stuff.
  webpack-dev-server - 4.7.4 -- Serves a webpack app. Updates the browser on changes.
  workbox-webpack-plugin - Not installed
  register-service-worker - 1.7.2 -- Script for registering service worker, with hooks
  typescript - 4.5.5 -- TypeScript is a language for application scale JavaScript development
  @capacitor/core - Not installed
  @capacitor/cli - Not installed
  @capacitor/android - Not installed
  @capacitor/ios - Not installed

Quasar App Extensions
  @quasar/quasar-app-extension-testing - 1.0.5 -- A Quasar App Extension for managing Test Harnesses
  @quasar/quasar-app-extension-testing-e2e-cypress - 3.1.2 -- A Quasar App Extension for Cypress e2e
  @quasar/quasar-app-extension-testing-quality - 1.0.0-beta.8 -- A Quasar App Extension for Code Quality
  @quasar/quasar-app-extension-testing-security - 1.0.0-alpha.8 -- A Quasar App Extension for Application Security, using OWASP ZAP
  @quasar/quasar-app-extension-testing-unit-jest - 3.0.0-alpha.10 -- A Quasar App Extension for running Jest tests

Networking
  Host - pop-os
  wlo1 - 192.168.0.196
```

**Comment by pdanpdan** [comment]
QInput is rendered when it is hidden, so the height cannot be computed.
The easiest way to solve it is to use a `key` on it to force re-render
https://codepen.io/pdanpdan/pen/eYVmYgp

**Comment by maarteNNNN** [comment]
@pdanpdan I tried that on the case that initially introduced the problem. It's a bit more complex however. Adding the `:key` didn't resolve it. However I wanted to know how Quasar manages the animation events? It seems like it should fire `onAnimationend` right? I tried adding a manual `document.body.addEventListener('animationend', () => console.log('ended'))` to see what happens but nothing is getting logged...
